src/main/java/com/minecolonies/coremod/entity/mobs/EntityMercenary.java

```diff
@@ -128,7 +131,8 @@ public EntityMercenary(final EntityType<EntityMercenary> type, final World world
 
         this.goalSelector.addGoal(0, new SwimGoal(this));
         this.goalSelector.addGoal(1, new EntityMercenaryAI(this));
-        this.goalSelector.addGoal(2, new NearestAttackableTargetGoal<>(this, MonsterEntity.class, false));
+        this.goalSelector.addGoal(2, new OpenDoorGoal(this, true));
```

Comment:
weren't they doing this anymore?

Reply:
not sure if it even works in 1.12 but in 1.14 they're stuck on doors

Reply:
You should probably also add the fence gate one (probably also for 1.12 then)